### PR TITLE
Export all specs from index.ts

### PR DESCRIPTION
**Pull request by mschrage**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Exports all specs as modules. (Autogenerated in index.js)
**What is the current behavior? (You can also link to an open issue here)**
It is challenging to use `@withfig/autocomplete` when installed as an `npm` package.
**What is the new behavior (if this is a feature change)?**
`import { git } from '@withfig/autocomplete'` or `import * as specs from '@withfig/autocomplete'`

**Additional info:**

The default export (a `string[]` of all completion files in `src/**.*.ts`) remains the same to ensure backwards compatibility

### CODE REVIEW

**Comment by mschrage** [comment]
@fedeci @QuiiBz @sullivan-sean this PR still doesn't produce a fully idiomatic TypeScript npm package since the `.d.ts` types aren't generated. Any thoughts on how to resolve this?

Screen Shot 2021-12-26 at 6 21 20 PM

**Comment by mschrage** [comment]
It might be as easy as generating a `.d.ts` file like this and referencing it in the `types` field of `package.json`:

```ts
declare module '@withfig/autocomplete' {
    export declare const git: Fig.Spec
    export declare const aws: Fig.Spec
    export declare const heroku: Fig.Spec
    ...
    const completions: string[]
    export default completions
}
```

File: generate-index.ts [hunk 1]
```diff
@@ -25,10 +25,26 @@ const files = glob
...
+const commandTypeDeclarations = commands
+  .map((x) => `export declare const ${normalize(x)}: Fig.Spec`)
+  .join("\n");
+
+fs.writeFileSync(
+  "build/index.d.ts",
+  `${commandTypeDeclarations}
+
+const completions: string[]
+export default completions
```
Comment (fedeci) [inline generate-index.ts:48]:
```suggestion
declare const completions: string[]
export { completions as default }
```

File: generate-index.ts [hunk 2]
```diff
@@ -25,10 +25,26 @@ const files = glob
...
+  `var e=[${files.join(",")}];export{e as default};${commandExports}`
+);
+
+const commandTypeDeclarations = commands
+  .map((x) => `export declare const ${normalize(x)}: Fig.Spec`)
+  .join("\n");
+
+fs.writeFileSync(
+  "build/index.d.ts",
+  `${commandTypeDeclarations}
```
Comment (fedeci) [inline generate-index.ts:45]:
```suggestion
  `import "@withfig/autocomplete-types"
  
${commandTypeDeclarations}
```

**Comment by fedeci** [comment]
We are exporting ESModules from the packages, so we also need to specify `type: "module"` in the `package.json`.

**Review by fedeci — APPROVED** [review]

**Comment by fedeci** [comment]
I don't think we need autocomplete-types as a dep, then we should be ready to merge it.

**Comment by miketheman** [comment]
It appears the [build&release step is still broken](https://github.com/withfig/autocomplete/runs/4751487793?check_suite_focus=true#step:4:11) as a result of this change.
I corrected the lint error in #821 , and poked at how to resolve the ts-node issue, but got lost in the mess of what `ts-node` is even trying to do, or updating versions, etc.

**Pull request merged**